src/caregiver/caregiverEdit.js

```diff
@@ -0,0 +1,206 @@
+import axiosInstance from "../api/axiosInstance";
+
+const CaregiverEdit = ({ isRegistered, onSuccess }) => {
+  const [formData, setFormData] = useState({
+    realName: "",
+    salary: "",
+    servNeeded: "",
+    loc: "",
+    employmentType: "CONTRACT",
+    workForm: "COMMUTE",
+    workTime: "MORNING",
+    workDays: "",
+    status: "OPEN",
+  });
+
+  const [workDaysInput, setWorkDaysInput] = useState("");
+
+  useEffect(() => {
+    const fetchCaregiverData = async () => {
+      try {
+        const response = await axiosInstance.get("/api/caregivers/user");
+        if (response.status === 200) {
+          setFormData(response.data);
+          setWorkDaysInput(convertBinaryToDays(response.data.workDays));
+        }
+      } catch (error) {
+        console.error("Caregiver 데이터를 불러오는 중 오류 발생:", error);
+      }
+    };
+
+    if (isRegistered) {
+      fetchCaregiverData();
+    }
+  }, [isRegistered]);
+
+  const convertDaysToBinary = (daysString) => {
+    if (!daysString) return "";
+    const days = ["월", "화", "수", "목", "금", "토", "일"];
+    return days.map((day) => (daysString.includes(day) ? "1" : "0")).join("");
+  };
+
+  const convertBinaryToDays = (binaryString) => {
+    const days = ["월", "화", "수", "목", "금", "토", "일"];
+    return binaryString
+        .split("")
+        .map((bit, index) => (bit === "1" ? days[index] : ""))
+        .join("");
+  };
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    if (name === "workDays") {
+      setWorkDaysInput(value);
+    } else {
+      setFormData({ ...formData, [name]: value });
+    }
+  };
+
+  const handleWorkDaysBlur = () => {
+    setFormData({
+      ...formData,
+      workDays: convertDaysToBinary(workDaysInput),
+    });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    try {
+      const endpoint = "/api/caregivers/build";
+        const method = "post";
+        const response = await axiosInstance[method](endpoint, formData);
+        if (response.status === 201) {
+            alert(`요양사 ${isRegistered ? "수정" : "등록"}이 완료되었습니다!`);
+            onSuccess();
+            window.location.reload();
+        }
+    } catch (error) {
+      console.error(isRegistered ? "요양사 업데이트 실패:" : "요양사 등록 실패:", error);
+      alert(isRegistered ? "요양사 정보 수정 중 오류가 발생했습니다." : "요양사 등록 중 오류가 발생했습니다.");
+    }
+  };
+
+  if (!formData) return <div>로딩 중...</div>;
+
+  return (
+    <div className={styles.editCaregiverPage}>
+      <h2>{isRegistered ? '요양사 정보 수정' : '요양사 등록'}</h2>
+      <form onSubmit={handleSubmit} className="space-y-4">
+        <div className={styles.inputGroup}>
+          <label>실명</label>
+          <input
+            type="text"
+            id="realName"
+            name="realName"
+            value={formData.realName}
+            onChange={handleChange}
+            required
+          />
+        </div>
+        <div className={styles.inputGroup}>
+          <label>희망 월급</label>
```

Comment:
```suggestion
          <label>희망 월급 (단위: 만 원)</label>
```

단위를 명확히 하면 좋을 것 같아요. 보통 월급은 만원 단위로 생각하니까 제가 임의로 만원이라고 적어봤어요.

혹시 아래같이 바꿔주실 수 있다면 이게 UI는 더 좋아보이네요.

![image](https://github.com/user-attachments/assets/3d4f41ec-3e51-4ace-a013-e482dda4bfb8)


이걸 생각해보니 또 데이터베이스에 어떻게 저장할지 고민이네요.
UI에선 n만원이라고 적더라도 데이터베이스에는 n보다는 n*10000 으로 저장하는게 낫겠죠?